### ISSUE TITLE
High Active contour loss for brain tumor segmentation

### ISSUE THREAD

**Issue by gunjannaik** [lc82111/Active-Contour-Loss-pytorch#2]
While using active contour loss on Brats data, I am getting high loss and good dice score. But, the output is blank after the prediction for validation data. Am I missing something? Do I have to combine categorical plus active contour loss?
![loss_issue](https://user-images.githubusercontent.com/14123491/72207562-5a34db00-34c0-11ea-8996-5929bca0423c.png)

**Comment by lc82111** [comment]
Yes, I also found your discoveries (high dice but bad prediction). You probably did not miss anything. I also tried the combination of cross-entropy and active contour loss, and observed an improvement. But it's crucial to use a very small weight to suppress the influence of active loss.

**Comment by gunjannaik** [comment]
Can you tell me the meaning of small weight ? Do I train it for few iterations, using cross-entropy and then use it as pretrained weights?

**Comment by lc82111** [comment]
1. The small weight means using a weight to balance the influence between CE and ActiveLoss. (i.e. CE+\labmda ActiveLoss, where \labmda is weight and should be small.)

2. No, I suggest you use CE+smoothness or CE alone, instead of Activeloss. The reason is: with some analysis, you will realize ActiveLoss is nothing more than a prediction_errors + a prediction smoothness. I suggest using CE instead of prediction_errors. I found CE is very stable.
